### PR TITLE
Create releases using GH actions

### LINKED ISSUE #2: Set up creation of GH releases using GH actions
Should be triggered on creation of a version tag - we want a draft release to be created to give us a chance to add release notes before publishing. Should include a tarball of the scripts, samples and templates.

### PR DESCRIPTION
**Pull request by legaul-cisco**
See https://github.com/ios-xr/xrd-tools/actions/workflows/release.yml

Verified contents of release is as expected and sha256sum verification succeeds.

Fixes #2

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          sha256sum "$filename" > "$filename".sha256
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: Release ${{ github.ref_name }}
+          body: |
+            Changes in this release
```
Comment (lawrencetroup) [inline .github/workflows/release.yml:33]: What's this `body` section used for? It currently has just some boilerplate changelog in.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            - First change
+            - Second change
+          files: xrd-tools-${{github.ref_name}}.tgz*
```
Comment (lawrencetroup) [inline .github/workflows/release.yml:36]: You don't need to list the `.sha256` file here too?

Comment (legaul-cisco) [reply .github/workflows/release.yml:36]: Uses glob at the end, would you prefer both files be listed explicitly?

File: .github/workflows/release.yml (hunk 1 above)
Comment (legaul-cisco) [reply .github/workflows/release.yml:33]: Exactly that :) it's boilerplate that hopefully makes it clear *something* should go there when going to publish the release. This action creates a draft release that should then manually have release notes added before being published.

File: .github/workflows/release.yml (hunk 2 above)
Comment (lawrencetroup) [reply .github/workflows/release.yml:36]: I think both files explicitly would be clearer (and then doesn't risk catching some other files that happen to match the glob).

**Review by lawrencetroup — APPROVED** [review]

**Pull request merged**